src/app/modules/form/components/input-text/input-text.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, Input } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+@Component({
+	selector: 'input-text',
+	templateUrl: './input-text.component.html',
+	styleUrls: ['./input-text.component.scss'],
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: InputTextComponent,
+			multi: true,
+		},
+	],
+})
+export class InputTextComponent implements ControlValueAccessor {
+	@Input() public label!: string;
+	public val = '';
+
+	public onChange!: (value: string) => void;
+	public onTouch!: () => void;
+
+	public writeValue(value: string): void {
+		this.val = value;
+	}
+	public registerOnChange(fn: (value: string) => void): void {
+		this.onChange = fn;
+	}
+	public registerOnTouched(fn: any): void {
+		this.onTouch = fn;
+	}
+
+	public handleInput($MouseEvent: any): void {
```

Comment:
...
public handleInput(event: MouseEvent): void {
   if(event.target instanceof HTMLInputElement) {
     this.val = $MouseEvent.target.value;
     this.onChange(this.val);
     this.onTouch();
   }
}